core/templates/dev/head/pages/exploration_player/FeedbackPopupDirective.js

```diff
@@ -108,8 +108,13 @@ oppia.directive('feedbackPopup', [
 
           $scope.saveFeedback = function() {
             if ($scope.feedbackText) {
+              var feedbackSubject = $scope.feedbackText.substr(0, 60);
```

Comment:
This needs tests. Please pull it out into a separate function/service, something like getAbbreviatedFeedbackText(), and add Karma tests for it to ensure that the various cases are correctly handled.

Reply:
I don't understand why is there a need for tests in something like 'substr()'?

Reply:
Sorry, I don't mean just substr(). I mean the whole process of generating a subject string from a text message.

Reply:
@seanlip I have not written any tests before so could you guide me a bit here.

Reply:
I suggest looking at the existing Karma tests, in files named *Spec.js. You'll want to extract this code into a function in its own service, or into a filter and then write tests for that function/filter.

@kevinlee12 should be able to provide more guidance, too, but you should be able to get started by looking at the existing code. There's also some guidance on [this wiki page](https://github.com/oppia/oppia/wiki/Running-Tests#client-side-tests).